asm-to-pil/src/romgen.rs

```diff
@@ -146,38 +140,33 @@ pub fn generate_machine_rom<T: FieldElement>(
             let input_substitution = function
```

Comment:
`.params.inputs.params.iter()` or similar call chains appear very often in this code. Could we have a functions `input_params` / `output_params` that return an iterators or something? (could be done in another pr)